staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -269,37 +309,56 @@ func (cfgCtlr *configController) processNextWorkItem() bool {
 
 	func(obj interface{}) {
 		defer cfgCtlr.configQueue.Done(obj)
-		if !cfgCtlr.syncOne() {
+		report := cfgCtlr.SyncOne(map[string]string{})
+		switch {
+		case report.NeedRetry:
 			cfgCtlr.configQueue.AddRateLimited(obj)
-		} else {
+		case report.NeededSpecificWait > 0:
+			cfgCtlr.configQueue.AddAfter(obj, report.NeededSpecificWait)
+		default:
 			cfgCtlr.configQueue.Forget(obj)
 		}
 	}(obj)
 
 	return true
 }
 
-// syncOne attempts to sync all the API Priority and Fairness config
-// objects.  It either succeeds and returns `true` or logs an error
-// and returns `false`.
-func (cfgCtlr *configController) syncOne() bool {
+// SyncReport describes the outcome of a call to SyncOne
+type SyncReport struct {
+	// Need to wait this much time for the sake of rate limiting
+	// controller fights
+	NeededSpecificWait time.Duration
+
+	// NeedRetry indicates whether to requeue with exponential backoff
+	NeedRetry bool
+}
+
+// SyncOne does one full synchronization.  It reads all the API
+// objects that configure API Priority and Fairness and updates the
+// local configController accordingly.
+// Only invoke this in the one and only worker goroutine
+func (cfgCtlr *configController) SyncOne(flowSchemaRVs map[string]string) SyncReport {
```

Comment:
why this has to be public?

Reply:
Added explanation in TestableInterface.

Reply:
I don't think it's really needed, because you need to call it from unit tests, which live in the same package, which means they can exercise also private methods.

Reply:
`SyncOne` is also called from integration test (https://github.com/MikeSpreitzer/kubernetes/blob/ratelimit/test/integration/apiserver/flowcontrol/fight_test.go)